R/rtf_rich_text.R

```diff
@@ -0,0 +1,116 @@
+#    Copyright (c) 2023 Merck & Co., Inc., Rahway, NJ, USA and its affiliates. All rights reserved.
+#
+#    This file is part of the r2rtf program.
+#
+#    r2rtf is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#' Text to formatted RTF Encode
+#'
+#' @param text Plain text.
+#' @param theme Named list defining themes for tags. See \code(rtf_text) for
+#' details on possible formatting.
+#'
+#' @section Specification:
+#' \if{latex}{
+#'  \itemize{
+#'    \item Validate if theme list items correspond to \code{font_type()} arguments.
+#'    \item Create regex expressions to match `{}` and `.tag` in text.
+#'    \item Extract tagged text from input text.
+#'    \item Extract tags from tagged text.
+#'    \item Extract text from tagged text.
+#'    \item Validate that lengths of extractions are all the same.
+#'    \item Validate that tags are defined in the `theme` argument.
+#'    \item Execute `rtf_text` with extracted text and relevant formatting.
+#'    \item Reinsert encoded formatted text to original input text.
+#'  }
+#'  }
+#' \if{html}{The contents of this section are shown in PDF user manual only.}
+#'
+#' @importFrom checkmate assert_choice assert_true
+#'
+#' @export
+#'
+#' @examples
+#' rtf_rich_text()
+rtf_rich_text <- function(text = "This is {.emph important}. This is {.blah relevant}.",
+                          theme = list(
+                            .emph = list(color = "blue", `format` = "b"),
+                            .blah = list(color = "red")
+                          )) {
+  # bulletproof theme
+  theme_args <- theme
+  names(theme_args) <- NULL
+  unique_theme_args <- unique(names(unlist(theme_args)))
+  purrr::walk(
```

Comment:
r2rtf has zero dependency. Could we refactor the code to avoid using `purrr` and `checkmate`? 

check if `r2rtf:::check_args` be sufficient.

Reply:
I'll look into it a bit